frontend/src/components/dialogs/SecretDialogGeneric.vue

```diff
@@ -0,0 +1,156 @@
+<!--
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+ -->
+
+ <template>
+  <secret-dialog
+    :value=value
+    :data="secretData"
+    :data-valid="valid"
+    :secret="secret"
+    :vendor="vendor"
+    :create-title="`Add new ${vendor} Secret`"
+    :replace-title="`Replace ${vendor} Secret`"
+    @input="onInput">
+
+    <template v-slot:secret-slot>
+      <div>
+        <v-textarea
+          ref="data"
+          color="primary"
+          filled
+          v-model="data"
+          label="Secret Data"
+          :error-messages="getErrorMessages('data')"
+          @input="$v.data.$touch()"
+          @blur="$v.data.$touch()"
+        ></v-textarea>
+      </div>
+    </template>
+    <template v-slot:help-slot>
+      <div class="help-content">
+        <p>
+          This is a generic provider service account dialog.
+        </p>
+        <p>
+          Please enter data required for this provider type.
```

Comment:
use vendor prop instead of "this provider type"?

Reply:
done